python/BaseRegTest.py

```diff
@@ -6,7 +6,7 @@
 
 class BaseRegTest(object):
 
-    def __init__(self, ref_file, train=False, comm=None):
+    def __init__(self, ref_file, train=False, comm=None, check_arch=False):
```

Comment:
@nwu63  Why make the default False?

Reply:
Because you only need to use this functionality if a package requires PETSc. I didn't want to change anything for the other packages, in case it introduced some different behaviour.

Reply:
ok, cool